README.md

```diff
@@ -6,7 +6,80 @@
 [![GitHub license](https://img.shields.io/github/license/golemfactory/yapapi)](https://github.com/golemfactory/yapapi/blob/master/LICENSE)
 [![GitHub issues](https://img.shields.io/github/issues/golemfactory/yapapi)](https://github.com/golemfactory/yapapi/issues)
 
-## How to use
+## What's Golem, btw?
+
+[Golem](https:://golem.network) is a global, open-source, decentralized supercomputer 
+that anyone can access. It connects individual machines - be that laptops, home PCs or 
+even data centers - to form a vast network, the purpose of which is to provide a way to 
+distribute computations to its provider nodes and allow requestors to utilize its unique 
+potential - which can lie in its combined computing power, the geographical distribution 
+or its censorship resistance.
+
+## Golem's requestor setup
+
+Golem's requestor-side configuration consists of two separate components:
+* the [`yagna` daemon](https://github.com/golemfactory/yagna) - your node in the 
+  new Golem network, responsible for communication with the other nodes, running the 
+  market and providing easy access to the payment mechanisms.
+* the requestor agent - the part that the developer of the specific Golem application
+  is responsible for.
+
+The daemon and the requestor agent communicate using three REST APIs which 
+`yapapi` - Golem's Python high-level API - aims to abstract to large extent to make 
+application development on Golem as easy as possible.
+
+## How to use this API?
+
+Assuming you have your Golem node up and running (you can find instructions on how to 
+do that in the [yagna repository](https://github.com/golemfactory/yagna) and in our 
+[handbook](https://handbook.golem.network)), what you need to do is:
+* *prepare your payload* - this needs to be a Docker image containing your application
```

Comment:
```suggestion
* *Prepare your payload* - this needs to be a Docker image containing your application
```
Not sure about this.

Reply:
same here actually... 

logically, it's part of the sentence above ...